dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -1371,36 +1374,39 @@ void dpnp_rng_vonmises_small_kappa_c(void* result, const _DataType mu, const _Da
 
     Uvec = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(size * sizeof(_DataType)));
     Vvec = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(size * sizeof(_DataType)));
+    n = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(sizeof(size_t)));
 
-    for (size_t n = 0; n < size;)
+    for (*n = 0; *n < size;)
     {
-        size_t diff_size = size - n;
+        size_t diff_size = size - *n;
         mkl_rng::uniform<_DataType> uniform_distribution_u(d_zero, M_PI);
-        auto event_out = mkl_rng::generate(uniform_distribution_u, DPNP_RNG_ENGINE, diff_size, Uvec);
-        event_out.wait();
-        // TODO
-        // use deps case
+        auto uniform_distr_u_event = mkl_rng::generate(uniform_distribution_u, DPNP_RNG_ENGINE, diff_size, Uvec);
         mkl_rng::uniform<_DataType> uniform_distribution_v(d_zero, d_one);
-        event_out = mkl_rng::generate(uniform_distribution_v, DPNP_RNG_ENGINE, diff_size, Vvec);
-        event_out.wait();
+        auto uniform_distr_v_event = mkl_rng::generate(uniform_distribution_v, DPNP_RNG_ENGINE, diff_size, Vvec);
 
-        // TODO
-        // kernel
-        for (size_t i = 0; i < diff_size; i++)
-        {
-            _DataType Z, W, Y, V;
-            Z = cos(Uvec[i]);
-            V = Vvec[i];
-            W = (kappa + s_kappa * Z) / (s_kappa + kappa * Z);
-            Y = s_kappa - kappa * W;
-            if ((Y * (2 - Y) >= V) || (log(Y / V) + 1 >= Y))
-            {
-                result1[n++] = acos(W);
-            }
-        }
-    }
+        cl::sycl::range<1> diff_gws((diff_size));
 
+        auto paral_kernel_some = [&](cl::sycl::handler& cgh) {
+            cgh.depends_on({uniform_distr_u_event, uniform_distr_v_event});
+            cgh.parallel_for(diff_gws, [=](cl::sycl::id<1> global_id) {
+                size_t i = global_id[0];
+                _DataType Z, W, Y, V;
+                Z = cl::sycl::cos(Uvec[i]);
+                V = Vvec[i];
+                W = (kappa + s_kappa * Z) / (s_kappa + kappa * Z);
+                Y = s_kappa - kappa * W;
+                if ((Y * (2 - Y) >= V) || (cl::sycl::log(Y / V) + 1 >= Y))
+                {
+                    *n = *n + 1;
+                    result1[*n] = cl::sycl::acos(W);
```

Comment:
Looks like here we are getting race condition, that is why we are getting wrong results. To prevent it we should calculate n (index of result) from i.